### PR TITLE
adds request parser tests

### PR DESCRIPTION
**Pull request by crystalhsieh**

### CODE REVIEW

File: connection_test.cc [hunk 1]
```diff
@@ -9,10 +9,19 @@
...
+  }
+  
+  // Inject the maps created before connection is initialized.
+  void injectTestMaps() {
+    echo_map["/echo"] =  "";
+    static_map["/static"] = "static_files";
   }
 
   boost::asio::io_service io_service_;
+  std::map<std::string, std::string> echo_map;
```
Comment (ryanvoong) [inline connection_test.cc]: member variables should have trailing underscore

File: request_parser_test.cc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  void setUpRequest(std::string buffer) {
+    test_buffer = buffer;
+    runParse();
+  }
+
+  void properRequest() {
+    setUpRequest("GET /echo/hello/there.html HTTP/1.1\r\n" + '\0');
+  }
+
+  std::string test_buffer;
```
Comment (ryanvoong) [inline request_parser_test.cc]: member variables should have trailing underscore

File: request_parser_test.cc [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#define TEST_REQUEST_PARSER
+  
+#include "gtest/gtest.h"
+#include "request_parser.h"
+#include "request.h"
+
+#define BUFFER_SIZE 1024 
+
+class RequestParserTest : public ::testing::Test {
+  protected:
```
Comment (ryanvoong) [inline request_parser_test.cc]: should be indented only 1 space

**Review by ryanvoong — COMMENTED** [review]
looks good

File: connection_test.cc (hunk 1 above)
Comment (crystalhsieh) [reply connection_test.cc]: done

File: request_parser_test.cc (hunk 3 above)
Comment (crystalhsieh) [reply request_parser_test.cc]: done

**Review by ryanvoong — APPROVED** [review]

**Pull request merged**